### PR TITLE
Ported aten::cross to work with symints

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82209
* #82058
* #82054
* __->__ #82052
* #82051
* #82049

### CODE REVIEW

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -441,7 +441,7 @@ def f(a, b):
     xfail('column_stack', ''),  # Tensors of type TensorImpl do not have numel
     xfail('constant_pad_nd', ''),  # aten.fill.Scalar - couldn't find symbolic meta function/decomposition
     xfail('count_nonzero', ''),  # Could not run 'aten::count_nonzero.dim_IntList' with arguments from the 'Meta' ba...
-    xfail('cross', ''),  # aten.size.default - couldn't find symbolic meta function/decomposition
+    xfail('cross', ''),  # aten.linalg_cross.default - couldn't find symbolic meta function/decomposition
```
Comment (Chillee) [inline test/test_proxy_tensor.py:444]: hell yeah, all this work to make it fail on a different function :)

File: c10/core/SymInt.h [hunk 2]
```diff
@@ -39,18 +39,13 @@ class C10_API SymInt {
...
-
-  bool operator!=(const SymInt& p2) const {
-    return data_ != p2.data_;
-  }
-
   SymInt operator+(SymInt sci) const;
   SymInt operator*(SymInt sci) const;
   bool operator<(SymInt sci) const;
   void operator*=(SymInt sci);
+  // Perhaps these should return symbolic bools?
```
Comment (ezyang) [inline c10/core/SymInt.h:46]: would be nice, but having the immediately guard is probably not too bad for now.

File: c10/core/SymInt.cpp [hunk 3]
```diff
@@ -29,23 +51,20 @@ SymInt SymInt::operator*(SymInt sci) const {
...
+  if (!is_symbolic() && !sci.is_symbolic()) {
+    return data_ == sci.data_;
   }
-  return SymInt::toSymInt(a->add(b));
+  auto res = normalize_symints(*this, sci);
+  return res[0]->eq(res[1])->bool_();
+}
+
+bool SymInt::operator!=(SymInt sci) const {
+  return !(*this == sci);
```
Comment (ezyang) [inline c10/core/SymInt.cpp:67]: I told @Krovatkin that we needed to define neq directly, but now I've forgotten why

**Review by ezyang — APPROVED** [review]

File: c10/core/SymInt.cpp (hunk 3 above)
Comment (Chillee) [reply c10/core/SymInt.cpp:67]: Perhaps since torchinductor does some special simplifications for `==` vs. `!=`?

**Comment by eellison** [comment]
This breaks the build - https://github.com/pytorch/pytorch/runs/7531630216?check_suite_focus=true see https://hud.pytorch.org/ (this or one of the previous PRs)

**Comment by Chillee** [comment]
@eellison looks like this one.

**Pull request closed without merging**